### PR TITLE
Optimize AbandonedList Space Usage

### PR DESCRIPTION
**Pull request by dipkakwani**
Remove BatchIds from the Abandoned List as it is redundant information which is already stored in each Abandoned page. https://github.com/NethermindEth/Paprika/issues/410

### CODE REVIEW

File: src/Paprika/Store/AbandonedList.cs [hunk 1]
```diff
@@ -166,15 +158,13 @@ private void Register(DbAddress head, IBatchContext batch)
     {
         if (MaxCount == EntriesCount)
         {
-            // No place, attach it to the youngest batch id from the sorted BatchIds
+            // No place, attach it to the last page which has the youngest batch id. Addresses
+            // are always sorted by batch ids, hence the last page.
             var maxAt = MaxCount - 1;
-            Debug.Assert(MaxCount > 2 && BatchIds[MaxCount - 1] > BatchIds[MaxCount - 2]);
```
Comment (dipkakwani) [inline src/Paprika/Store/AbandonedList.cs:171]: Removed the debug assert to keep the code shorter. We can fetch the corresponding batch ids from the pages, but not sure if it is required?

Comment (Scooletz) [reply src/Paprika/Store/AbandonedList.cs:171]: I think it's ok now.

**Review by Scooletz — CHANGES_REQUESTED** [review]
One thing to fix. 
Also, maybe a test for it? There was the chaining one, why not to use it again?

File: src/Paprika/Store/AbandonedList.cs [hunk 2]
```diff
@@ -22,10 +22,6 @@ public struct AbandonedList
     [FieldOffset(0)] private DbAddress Current;
     [FieldOffset(DbAddress.Size)] private uint EntriesCount;
 
-    [FieldOffset(EntriesStart)] private uint BatchIdStart;
```
[the remark this answers is not in the document]
Comment (dipkakwani) [reply src/Paprika/Store/AbandonedList.cs:25]: My bad, completely missed changing the offsets 😓 
Fixed it in the latest iteration and also added a sanity check on the MaxCount value (without exposing complete data structure details).

**Review by Scooletz — APPROVED** [review]
😍

**Pull request merged**